src/net/download.rs

```diff
@@ -118,6 +118,12 @@ pub const BLOCK_REREQUEST_INTERVAL: u64 = 30;
 /// This module is responsible for downloading blocks and microblocks from other peers, using block
 /// inventory state (see src/net/inv.rs)
 
+#[derive(Debug, PartialEq, Clone, Hash, Eq)]
```

Comment:
Please revert all changes to `src/net/download.rs`.

Reply:
Ha, can we keep this changes actually?
By having the `BlockRequestKey` conforming to the trait `Requestable` I can reuse the logic implemented in the `begin_request` method of the PeerNetwork for the attachments, instead of duplicating it.

Reply:
Okay, I annotated this file to describe the set of changes I'd like you to make in order to keep using `begin_request()` with the attachment downloader.  If you can do these things, then the changes can stay.